### PR TITLE
Add support for STM32F373 und F427 FSMC

### PR DESCRIPTION
**Pull request by dergraaf**
Adds support for the STM32F373 family and with that enables timers >15 for all affected STM32 devices.
Adds the FMC pin definitions for the STM32F427. Renames the FSMC to FMC for the STM32 devices with SDRAM support.

### CODE REVIEW

File: src/xpcc/architecture/platform/devices/stm32/stm32f100-c_r_v-8_b.xml [hunk 1]
```diff
@@ -66,7 +66,7 @@
     <driver type="i2c" name="stm32" instances="1,2"/>
     <driver type="spi" name="stm32" instances="1,2"/>
     <driver type="spi" name="stm32_uart" instances="1,2,3"/>
-    <driver type="timer" name="stm32" instances="1,2,3,4,6,7,15,16,17"/>
+    <driver type="timer" name="stm32" instances="1,2,3,4,6,7"/>
```
Comment (salkinium) [inline src/xpcc/architecture/platform/devices/stm32/stm32f100-c_r_v-8_b.xml]: Are you sure? The datasheet says these exist on F100 medium density.

File: src/xpcc/architecture/platform/driver/timer/stm32/general_purpose.cpp.in [hunk 2]
```diff
@@ -7,26 +7,30 @@
...
+%%     set irq_name = "1_TRG_COM_TIM11"
+%%   elif id == 12
+%%     set irq_name = "8_BRK_TIM12"
+%%   elif id == 13
+%%     set irq_name = "8_UP_TIM13"
+%%   elif id == 14 and not target is stm32f0
+%%     set irq_name = "8_TRG_COM_TIM14"
+%%   else
+%%     set irq_name = id
+%%   endif
```
Comment (salkinium) [inline src/xpcc/architecture/platform/driver/timer/stm32/general_purpose.cpp.in:39]: Ugh. ST, what are you doing?

File: src/xpcc/architecture/platform/devices/stm32/stm32f100-c_r_v-8_b.xml (hunk 1 above)
Comment (dergraaf) [inline src/xpcc/architecture/platform/devices/stm32/stm32f100-c_r_v-8_b.xml]: You are right, those should exists. I removed them due to compiler errors. But those where not due to missing timers but due to missnamed interrupts in driver.

**Comment by salkinium** [comment]
Thanks :+1:
Please squash and then I can merge.

**Pull request merged**